src/Fixer/ControlStructure/YodaStyleFixer.php

```diff
@@ -0,0 +1,609 @@
+                T_DIV_EQUAL,    // /=
+                T_DOUBLE_ARROW, // =>
+                T_GOTO,         // goto
+                T_LOGICAL_AND,  // and
+                T_LOGICAL_OR,   // or
+                T_LOGICAL_XOR,  // xor
+                T_MINUS_EQUAL,  // -=
+                T_MUL_EQUAL,    // *=
+                T_OR_EQUAL,     // |=
+                T_PLUS_EQUAL,   // +=
+                T_RETURN,       // return
+                T_SL_EQUAL,     // <<
+                T_SR_EQUAL,     // >>=
+                T_THROW,        // throw
+                T_XOR_EQUAL,    // ^=
+            ];
+
+            if (defined('T_POW_EQUAL')) {
+                $tokens[] = T_POW_EQUAL; // **=
+            }
+
+            if (defined('T_COALESCE')) {
+                $tokens[] = T_COALESCE; // ??
+            }
+        }
+
+        static $otherTokens = [
+            // bitwise and, or, xor
+            '&', '|', '^',
+            // ternary operators
+            '?', ':',
+            // assignment
+            '=',
+            // end of PHP statement
+            ',', ';',
+        ];
+
+        return $token->isGivenKind($tokens) || $token->equalsAny($otherTokens);
+    }
+
+    /**
+     * Checks whether the tokens between the given start and end describe a
+     * variable.
+     *
+     * @param Tokens $tokens The token list
+     * @param int    $index  The first index of the possible variable
+     * @param int    $end    The last index of the possible variable
+     *
+     * @return bool Whether the tokens describe a variable
+     */
+    private function isVariable(Tokens $tokens, $index, $end)
+    {
+        if ($end === $index) {
+            return $tokens[$index]->isGivenKind(T_VARIABLE);
+        }
+
+        // handle multiple braces around statement ((($a === 1)))
+        while (
+            $tokens[$index]->equals('(')
+            && $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index) === $end
+        ) {
+            $index = $tokens->getNextMeaningfulToken($index);
+            $end = $tokens->getPrevMeaningfulToken($end);
+        }
+
+        $expectString = false;
+        while ($index <= $end) {
+            $current = $tokens[$index];
+            if ($current->isComment() || $current->isWhitespace() || $tokens->isEmptyAt($index)) {
+                ++$index;
+
+                continue;
+            }
+
+            // check if this is the last token
+            if ($index === $end) {
+                return $current->isGivenKind($expectString ? T_STRING : T_VARIABLE);
+            }
+
+            if ($current->isGivenKind([T_LIST, CT::T_DESTRUCTURING_SQUARE_BRACE_OPEN, CT::T_DESTRUCTURING_SQUARE_BRACE_CLOSE])) {
+                return false;
+            }
+
+            $next = $tokens[$tokens->getNextMeaningfulToken($index)];
+
+            // self:: or ClassName::
+            if ($current->isGivenKind(T_STRING) && $next->isGivenKind(T_DOUBLE_COLON)) {
```

Comment:
In these `if` statements you check some expected token kinds and increment `$index` with hardcoded values, assuming the next token is at this position, which is not always true (you provided cases in tests). Wouldn't it be better to store the actual position of the next meaningful token and reuse it to increment `$index`?

Reply:
this is a direct copy of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/701/files#diff-fe2f2514a225293636fa7d1795d7a67dR153 (that is why there are some many authors in the fixer listed),
I always read it as, we found start of a statement that we know is at least X tokens, so lets skip the known amount, from there we check again in the loop. In the end the result should always be the same.

Reply:
I think I've updated following all your other suggestions.